File: string/Solution13.java
class Solution13 {
    public int romanToInt(String s) {
    	int[] digit1 = new int[]{1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000};
    	String[] digit2 = new String[]{"I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
    	int result = 0;
    loop:	for(int i = 0; i < s.length(); i++){
    		if(i + 1 < s.length()){
    			String cur = s.substring(i, i + 2);
    			for(int k = 0; k < digit1.length; k++){
    				if(cur.equals(digit2[k])){
    					result += digit1[k];
    					i = i + 1;
    					continue loop;
    				}
    			}
    		}
    		String cur = s.substring(i, i + 1);
    		for(int k = 0; k < digit1.length; k++){
    				if(cur.equals(digit2[k])){
    					result += digit1[k];
    					continue loop;
    				}
    			}
    	}
    	return result;
    }

    public static void main(String[] args){
    	Solution13 s = new Solution13();
    	System.out.println(s.romanToInt("MCMXCIV"));
	}
}

File: C++/Solution13.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution13 {
public:
    int romanToInt(string s) {
        vector<pair<int, string>> map = {
            {1, "I"}, {4, "IV"}, {5, "V"}, {9, "IX"}, {10, "X"}, {40, "XL"}, {50, "L"}, {90, "XC"}, 
            {100, "C"}, {400, "CD"}, {500, "D"}, {900, "CM"}, {1000, "M"}
        };
        int result = 0;
        int size = map.size();
        int i = 0;
        while(i < s.length()){
            for(int index = size - 1; index >= 0; index--){
                auto cur = map[index];
                if(cur.second.length() == 2){
                    if(i + 1 < s.length() && s.substr(i, 2) == cur.second){
                        result += cur.first;
                        i = i + 2;
                        break;
                    }
                }
                else{
                    if(s[i] == cur.second[0]){
                        result += cur.first;
                        i++;
                        break;
                    }
                }
            }
        }
        return result;
    }
};